app/models/discourse_single_sign_on.rb

```diff
@@ -161,13 +161,31 @@ def match_email_or_create_user(ip_address)
           )
         end
 
+        if profile_background_url.present?
```

Comment:
There should be test cases for changes made here. For reference, see: https://github.com/discourse/discourse/blob/6f40037ba551e6e495e6f51263940b574721e66a/spec/models/discourse_single_sign_on_spec.rb#L446-L552